aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StorageGetPresignedUrlOperation.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3.operation;
+
+import android.annotation.SuppressLint;
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.core.Consumer;
+import com.amplifyframework.storage.StorageException;
+import com.amplifyframework.storage.operation.StorageGetUrlOperation;
+import com.amplifyframework.storage.result.StorageGetUrlResult;
+import com.amplifyframework.storage.s3.request.AWSS3StorageGetPresignedUrlRequest;
+import com.amplifyframework.storage.s3.service.StorageService;
+import com.amplifyframework.storage.s3.utils.S3RequestUtils;
+
+import java.net.URL;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * An operation to retrieve pre-signed object URL from AWS S3.
+ */
+public final class AWSS3StorageGetPresignedUrlOperation
+        extends StorageGetUrlOperation<AWSS3StorageGetPresignedUrlRequest> {
+    private final StorageService storageService;
+    private final ExecutorService executorService;
+    private final Consumer<StorageGetUrlResult> onResult;
+    private final Consumer<StorageException> onError;
+
+    /**
+     * Constructs a new AWSS3StorageGetUrlOperation.
+     * @param storageService S3 client wrapper
+     * @param executorService Executor service used for running
+     *                        blocking operations on a separate thread
+     * @param request getUrl request parameters
+     * @param onSuccess Notified when URL is generated.
+     * @param onError Notified upon URL generation error
+     */
+    public AWSS3StorageGetPresignedUrlOperation(
+            @NonNull StorageService storageService,
+            @NonNull ExecutorService executorService,
+            @NonNull AWSS3StorageGetPresignedUrlRequest request,
+            @NonNull Consumer<StorageGetUrlResult> onSuccess,
+            @NonNull Consumer<StorageException> onError
+    ) {
+        super(request);
```

Comment:
Later there is `getRequest().weBelieveItsNotNull()` so if super doesn’t require non-bull, this constructor should